### PR TITLE
Simplified code and remove unsed code

### PR DESCRIPTION
**Pull request by anaxita**
## Describe your feature

- [x] this PR was checked by `golangci-linter`
- [x] new code **did not** break current interfaces
- [x] all comments were written on **american english** language
- [x] all commit messages are spiced up with emojis, which wa defined in CG and in `.gitmessage`

This PR improve code by removing unused code, panic and implements one of TODO

## Main purpose
- Removed a panic in switch cases (return an error insead)
- Implemented response channels closing on reconnect
- Simplified a func for message id generating
- Removed unused func

## Possible issues

@quenbyako, please add these labels: `label1`,`label2`,`label3`

### CODE REVIEW

File: internal/utils/utils.go [hunk 1]
```diff
@@ -9,21 +9,13 @@ package utils
...
 	"crypto/sha1"
-	"fmt"
 	"math/rand"
-	"os"
-	"runtime"
 	"time"
 )
 
-// GenerateMessageId отдает по сути unix timestamp но ужасно специфическим образом
-// TODO: нахуя нужно битовое и на -4??
```
Comment (xepozz) [inline internal/utils/utils.go:20]: btw have you got the idea? 😄

File: mtproto.go [hunk 2]
```diff
@@ -362,8 +363,8 @@ messageTypeSwitching:
 		// игнорим, пришло и пришло, че бубнить то
 
 	case *objects.BadMsgNotification:
-		pp.Println(message)
-		panic(message) // for debug, looks like this message is important
+		_, _ = pp.Println(message)
```
Comment (xepozz) [inline mtproto.go:366]: There is no need to unpack

**Review by xepozz — CHANGES_REQUESTED** [review]

**Comment by xepozz** [comment]
@quenbyako could you process this pr?

**Comment by anaxita** [comment]
> @quenbyako could you process this pr?

Yes of course

File: internal/utils/utils.go (hunk 1 above)
Comment (anaxita) [reply internal/utils/utils.go:20]: > btw have you got the idea? smile

I really don't have an idea. It seems like code from indian stackoverflow ))

**Comment by anaxita** [comment]
up theme

**Review by quenbyako — CHANGES_REQUESTED** [review]
@anaxita really sorry for such a big delay, but could you please write some tests for this message id generator? I'm not sure that it won't break anything. This ialgorithm i found in tdlib, so who knows, maybe they are right...

**Pull request closed without merging**